pkg/counter/cache.go

```diff
@@ -239,12 +262,19 @@ func initialize(cl client.Client, toolchainStatus *toolchainv1alpha1.ToolchainSt
 	// initialize the cached counters from the UserSignup and MasterUserRecord resources.
 	_, masterUserRecordsPerDomainMetricExists := toolchainStatus.Status.Metrics[toolchainv1alpha1.MasterUserRecordsPerDomainMetricKey]
 	_, usersPerActivationMetricKeyExists := toolchainStatus.Status.Metrics[toolchainv1alpha1.UsersPerActivationMetricKey]
+	_, usersPerActivationAndDomainMetricKeyExists := toolchainStatus.Status.Metrics[toolchainv1alpha1.UserSignupsPerActivationAndDomainMetricKey]
 	if toolchainStatus.Status.HostOperator == nil ||
 		toolchainStatus.Status.HostOperator.MasterUserRecordCount == 0 ||
 		!usersPerActivationMetricKeyExists ||
+		!usersPerActivationAndDomainMetricKeyExists ||
 		!masterUserRecordsPerDomainMetricExists {
 		return initializeFromResources(cl, toolchainStatus.Namespace)
 	}
+	// Migration for CRT-1072: force initialization from existing UserSignups/MasterUserRecords
+	// TODO: remove as part of CRT-1074
+	if _, exists := toolchainStatus.Status.Metrics[toolchainv1alpha1.MasterUserRecordsPerDomainMetricKey]; !exists {
+		return initializeFromResources(cl, toolchainStatus.Namespace)
+	}
```

Comment:
Why do we need this part? I'm confused because:
1. This PR is related to `UserSignupsPerActivationAndDomainMetricKey` and not `MasterUserRecordsPerDomainMetricKey`
2. the presence of `MasterUserRecordsPerDomainMetricKey` is already checked in the previous if statement, so if it's missing then it will be initialized anyway

Reply:
right, this statement is useless and I'm not sure why it landed in here 😳
probably a left-over from when I rebased this PR against https://github.com/codeready-toolchain/host-operator/pull/437 since I worked on both PRs at the same time at some point 😬
Anyways, I removed it in 4b9b70d